lib/notification.go

```diff
@@ -10,6 +10,15 @@ type (
 	}
 )
 
+// NewNotification initializes and returns a Notification pointer
+func NewNotification(format NotificationFormat, payload []byte) (*Notification, error) {
+	if !format.IsValid() {
```

Comment:
I think it is better to use isTrue whenever possible because of readability. So my suggestion is to let the returns change place and do format.IsValid() instead.

Reply:
"Avoid nesting by handling errors first"😎

Reply:
Error should always be last in Go. So I think I'll keep it as it is